compiler/rustc_hir_typeck/src/errors.rs

```diff
@@ -172,3 +172,68 @@ impl AddToDiagnostic for TypeMismatchFruTypo {
         );
     }
 }
+
+#[derive(Diagnostic)]
+#[diag(hir_typeck_lang_start_incorrect_number_params)]
+#[note(hir_typeck_lang_start_incorrect_number_params_note_expected_count)]
+#[note(hir_typeck_lang_start_expected_sig_note)]
+pub struct LangStartIncorrectNumberArgs {
+    #[primary_span]
+    pub params_span: Span,
+    pub found_param_count: usize,
+}
+
+#[derive(Diagnostic)]
+#[diag(hir_typeck_lang_start_incorrect_param)]
+pub struct LangStartIncorrectParam<'tcx> {
+    #[primary_span]
+    param_span: Span,
+
+    #[suggestion(style = "short", code = "{expected_ty}")]
```

Comment:
You can include the applicability in this attribute, e.g. `#[suggestion(applicability = "machine-applicable", ...)] - might let you remove the constructors.